### ISSUE TITLE
Unable to use scroll upon page down button

### ISSUE THREAD

**Issue by Automagication** [rintoj/angular2-virtual-scroll#213]
I have added the "virtual-scroll" tag around my div tags.
Scrolling is working fine upon dragging the scroll bar using mouse however scroll through
"Page down button" is taking me back to top of the page.

Is there a solution for this.

**Comment by speige** [comment]
PageUp/Down are working for me in my app. It may be something app-specific.

If you're using a full-window scrollbar I think this works by default.
If the scroll area is only around a smaller section of the page rather than full window you probably have to focus the scroll area first by clicking inside it before hitting pageUp/down.

Maybe you could create a stackblitz to help troubleshoot the issue?

Without seeing your app, I don't know the root cause, but one possible solution would be to try The "parentScroll" option of the virtual-scroll component. For example: setting it window allows you to have a full-window scrollbar.

**Comment by Automagication** [comment]
Exactly, it works fine if i click on the smaller section of the page and hit page down.
How can i fix this?

**Comment by speige** [comment]
You have to use the parentScroll option. Read the documentation here on github. Also, there's an example of it in the "demo" folder if you look at the source code. The updated demo hasn't been published to http://rintoj.github.io/angular2-virtual-scroll yet, so you have to run it from your own box to see it in action, or just look at the source/documentation & copy it into your app.

**Comment by Automagication** [comment]
The scroll area is only around a **smaller section**, as you mentioned, that is causing the issue.Is there an option to enable this?

**Comment by speige** [comment]
It is a known issue that when the scroll area is a small section you have to put focus on that smaller section before pageUp/down will work. I do not know how to fix this. If someone else figures out a way, I'll accept a pull request. The only known workaround is to convert it to a full-window scroller by setting parentScroll to window.

**Comment by PapaNappa** [comment]
I found a workaround:

Add `tabindex` attribute to `virtual-scroller` (or the parent scroll element if you have one)
Add a click listener on the `virtual-scroller`, on which it sets the focus for the scroller/parent scroll element.

Template:
```
<virtual-scroller #scroll (click)="scrollClick()" tabindex="-1">
  …
</virtual-scroller>
```

Typescript:
```
  @ViewChild('scroll', { read: ElementRef })
  private scrollElement!: ElementRef;

  scrollClick() {
    (this.scrollElement.nativeElement as HTMLElement).click();
  }
```

Of course the caveat is if you rely on elements within the scroller to have focus, this does not work.

**Comment by matte00** [comment]
I have same issues
